From: Vlastimil Babka <vbabka@suse.cz>
Subject: kabi: reorder new slabinfo fields in struct kmem_cache_node
Patch-mainline: never (kabi)
References: bnc#1116653

Hide the new fields total_slabs and free_slabs in struct kmem_cache_node from
the KABI checker, and also reorder them so that we quickly catch any 3rd party
module accessing this structure, as it's expected that if anyone does this
(which itself is unexpected), they would primarily access the list_heads.

Signed-off-by: Vlastimil Babka <vbabka@suse.cz>

---
 mm/slab.h |    6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

--- a/mm/slab.h
+++ b/mm/slab.h
@@ -327,11 +327,13 @@ struct kmem_cache_node {
 	spinlock_t list_lock;
 
 #ifdef CONFIG_SLAB
+#ifndef __GENKSYMS__
+	unsigned long total_slabs;	/* length of all slab lists */
+	unsigned long free_slabs;	/* length of free slab list only */
+#endif
 	struct list_head slabs_partial;	/* partial list first, better asm code */
 	struct list_head slabs_full;
 	struct list_head slabs_free;
-	unsigned long total_slabs;	/* length of all slab lists */
-	unsigned long free_slabs;	/* length of free slab list only */
 	unsigned long free_objects;
 	unsigned int free_limit;
 	unsigned int colour_next;	/* Per-node cache coloring */
